### PR TITLE
feat(agent): elapsed time + per-tool elapsed in Thinking indicator (#731 PR2)

### PR DESCRIPTION
**Pull request by isamu**
## Summary

- 既存の "Thinking…" インジケーター隣に **run 全体の経過時間** (\`· 1m 23s\`) と **実行中ツール毎の経過時間** (\`· 2.3s\`) を追加。長時間タスクが動いていることが目視できる。
- バックエンド・型・API スキーマ・i18n 全て**未変更**。フロント UI のみ。
- 新コンポーザブル \`useRunElapsed\` (1s 刻み) + 既存 \`usePendingCalls\` の返却値を per-tool elapsed 込みに narrow projection。
- 新 pure helper \`formatElapsed(ms)\` で magnitude-adaptive 表記 (\`0.3s\` / \`12s\` / \`1m 23s\` / \`1h 5m\`)。

## Items to Confirm / Review

- **副作用範囲ゼロ確認**: server / schema / type 変更は本当にゼロか? \`grep\` で既存 \`pendingCalls\` 利用箇所が ToolResultsPanel.vue 1ヶ所だけであることは確認済 (App.vue → ToolResultsPanel.vue → 表示のみ)。
- **1s tick の cost**: アクティブ session 1つにつき 1個の setInterval。ほぼ無視できるが、tab 非アクティブ時も走り続ける選択をしている (rAF だと throttle されて経過カウンタが止まるため)。
- **runElapsedMs >= 1000 ガード**: 1秒未満は表示しない。素早く返ってくる返答で 0s が一瞬光るのを防止。
- **per-tool elapsed の駆動**: 既存の 50ms displayTick (usePendingCalls) を流用。新規 timer なし。
- **format 整数 floor**: 12.999s → "12s" にする (round だと badge が秒針より先に進むのでユーザの体感と合わない)。
- **負の値の clamp**: stale tick race の防御で \`-X\` 入力は \`0.0s\` に丸める。
- **テスト**: \`formatElapsed\` 14ケース (各 magnitude band / boundary / floor / clamp) + \`useRunElapsed\` 7ケース (state transitions / immediate-watch / double-start guard / teardown idempotency)。

## User Prompt

> 731の続きってなにかできる？副作用内レベルで。
> AB セット実装して閉じよう。

(B 相当の "実行中ツール名表示" は実は既に \`pendingCalls\` リストとして存在していたので、本 PR では polish として **per-tool 経過時間を追加**。A の "経過時間" は新規実装。実質これで PR2 のスコープが完了。)

## Implementation

| File | Change |
|---|---|
| \`src/utils/agent/formatElapsed.ts\` (new) | Pure helper. 4段階の magnitude band、defensive negative clamp。|
| \`src/composables/useRunElapsed.ts\` (new) | \`isRunning\` watcher + 1s setInterval。\`{elapsedMs, teardown}\` を返す。 |
| \`src/composables/usePendingCalls.ts\` | 返却 shape を \`{toolUseId, toolName, elapsedMs}[]\` に narrow projection。50ms tick が既に動いているので elapsed が自動的に追従する。 |
| \`src/components/ToolResultsPanel.vue\` | Thinking indicator の \<div\> に run-level + per-tool の elapsed badge を追加。\`runElapsedMs\` prop を追加。 |
| \`src/App.vue\` | \`useRunElapsed\` を呼び出し \`runElapsedMs\` を ToolResultsPanel に渡す。teardown を既存の \`useEventListeners\` cleanup と合体。 |
| \`test/utils/agent/test_formatElapsed.ts\` (new) | 14 cases。 |
| \`test/composables/test_useRunElapsed.ts\` (new) | 7 cases (effectScope ベース)。 |

Plan: \`plans/feat-agent-elapsed-time-731.md\`。

## Test plan

- [x] \`yarn format\` clean
- [x] \`yarn lint\` (only pre-existing v-html warnings, 0 errors)
- [x] \`yarn typecheck\`
- [x] \`yarn build\`
- [x] \`yarn test\` — 3062/3062 pass (+20 new tests)
- [ ] CI lint_test matrix (Node 22.x / 24.x × ubuntu / macos / windows)
- [ ] CI e2e (Playwright)
- [ ] Manual: 長時間 agent run で経過時間が秒単位で更新されること確認
- [ ] Manual: ツール実行中の per-tool elapsed が更新されること確認
- [ ] Manual: 1秒未満で完了するクイック応答で run-level badge が出ないこと確認

## #731 Closing Plan

このPRがマージされたら #731 をクローズ:
- ✅ PR1 (#786): Stop button
- ✅ PR2 (本 PR): elapsed time + per-tool elapsed
- ⚠️ PR3 (将来): mid-flight 追加指示 — message-queue のバックエンド変更が必要なので別 issue で改めて検討

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Display total agent run elapsed time and per-tool elapsed time during long-running agent execution
  * Show elapsed time badges in the UI alongside pending tool operations
  * Support automatic time formatting with human-readable intervals (seconds, minutes, hours)

* **Tests**
  * Added comprehensive test coverage for elapsed time tracking and formatting logic to ensure accuracy and edge cases

### CODE REVIEW

File: src/utils/agent/formatElapsed.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const ONE_SECOND_MS = 1000;
+const ONE_MINUTE_MS = 60 * ONE_SECOND_MS;
+const ONE_HOUR_MS = 60 * ONE_MINUTE_MS;
+
+export function formatElapsed(elapsedMs: number): string {
+  // Defensive: clamp negatives to zero so a clock-skew or stale-tick
+  // race never renders "-0s".
+  const elapsed = elapsedMs > 0 ? elapsedMs : 0;
+  if (elapsed < ONE_SECOND_MS) {
```
Comment (isamu) [inline src/utils/agent/formatElapsed.ts:18]: This rounds sub-second durations up, so values in the ~950-999ms range will render as `1.0s` before the one-second boundary. If the intent is to never read ahead of the clock, floor the decimal or cap the sub-second branch instead of using `toFixed(1)` here.

File: src/composables/useRunElapsed.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    // flight (mounted mid-stream) starts ticking right away.
+    { immediate: true },
+  );
+
+  const elapsedMs = computed<number | null>(() => {
+    if (startedAt.value === null) return null;
+    return now.value - startedAt.value;
+  });
+
+  function teardown(): void {
```
Comment (isamu) [inline src/composables/useRunElapsed.ts:65]: `teardown()` only clears the current interval; the `watch()` stays live, so a later `isRunning = true` while this scope is still mounted will recreate the timer. That means this cleanup is not leak-proof on its own. Returning the watch stop handle or registering `onScopeDispose()` would make the composable self-contained.

**Review by isamu — COMMENTED** [review]
Reviewing the elapsed-time UI pass. I found two concrete issues that should be addressed before merge.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED

- `formatElapsed` rounds sub-second durations up, so values just under 1s can render as `1.0s` before the boundary.
- `useRunElapsed` cleanup is not scope-safe yet: clearing the interval without stopping the watcher lets the timer recreate itself if `isRunning` flips again before unmount.
- Non-blocking follow-ups: expose the Thinking row as a polite live region for screen readers, and localize the `s/m/h` suffixes if this UI is expected to follow the active locale.

I did not find any consumer breakage from the `usePendingCalls` shape change, and the `runElapsedMs >= 1000` gate looks correct.

**Comment by isamu** [comment]
## Iteration 1 fixes (commit c9535891)

Both MUST-FIX findings addressed. Summary:

**1. useRunElapsed teardown scope-safety** ✅
- Captured `WatchStopHandle` from `watch()`, call it first in `teardown()` before clearing the interval
- Regression test: flip `isRunning` after teardown, assert `elapsedMs` stays null

**2. formatElapsed sub-second floor-not-round** ✅
- `Math.floor(elapsed / 100) / 10` instead of `toFixed(1)` half-up rounding
- `999ms` → `0.9s` (was `1.0s`); `950ms` → `0.9s` test added to pin the half-up edge

**3. role="status" + aria-live="polite" on Thinking indicator** ✅ (in-scope a11y polish)

**Deferred** (your follow-up note): localizing `s/m/h` suffixes. Sits next to localized `statusMessage` but the unit-string lockstep across 8 locales is a separate i18n task — not in scope for #731 PR2. Happy to file a follow-up issue if the team agrees the cost is worth it.

Local: format/lint/typecheck/build green, 3064/3064 tests pass.

**Comment by isamu** [comment]
## Iteration 2 — Codex verdict: LGTM

Codex iter-2 review on `c9535891` returned LGTM (couldn't post itself due to sandbox network block — relaying):

> Both MUST-FIX items look closed in c9535891:
> - useRunElapsed.ts now stores and invokes the watch stop handle in teardown(), so a later isRunning flip cannot recreate the timer.
> - formatElapsed.ts now floors sub-second values to tenths, so 999ms stays 0.9s and the 0.0s/0.1s boundary behaves correctly.
>
> I didn't find a new regression in the teardown or rounding edges.

**Claude's eval**: clear. No remaining concerns from either reviewer. Local: format/lint/typecheck/build green, 3064/3064 tests pass.

Waiting on CI.
